dbmigrate-lib/src/drivers/sqlcipher.rs

```diff
@@ -0,0 +1,124 @@
+use url::Url;
+
+use sqlcipher_client::Connection;
+
+use super::Driver;
+use errors::{Result, ResultExt};
+
+const SQLCIPHER_DEFAULT_CIPHER: &'static str = "AES-256-CBC";
+
+#[derive(Debug)]
+pub struct Sqlcipher {
+    conn: Connection,
+}
+
+impl Sqlcipher {
+    pub fn new(url: &str) -> Result<Sqlcipher> {
+        // Expect a URL in the form:
+        // sqlcipher://user:password@host/path/to/file.db
+        //
+        // The user and host components are not used.
+        // The password is used with the PRAGMA KEY statement.
+        let parsedurl = Url::parse(url)
+            .chain_err(|| format!("Invalid SQLCipher URL: {}", url))?;
+        let path = parsedurl.path();
+        let conn = Connection::open(path)?;
+        let sqlcipher = Sqlcipher { conn: conn };
+        match parsedurl.password() {
+            Some(password) => {
+                sqlcipher.unlock(password)?;
+                sqlcipher.verify_cipher()?;
+                sqlcipher.ensure_migration_table_exists();
+            }
+            None => bail!("No password found in URL"),
+        }
+        Ok(sqlcipher)
+    }
+
+    fn unlock(&self, password: &str) -> Result<()> {
```

Comment:
can you add a comment on what this actually does for people not familiar to sqlcipher?

Reply:
Yes, I can add a doc-comment for the `unlock` method